### PR TITLE
feat(commits): added linter to enforce conventional commit messages

### PR DESCRIPTION
**Pull request by kleyow**
Opening up the discussion here.

I've included the default commitlint config. Any opinionated changes that anyone suggested be changed?

Only one I have is subject start always be capitalized. (I'm not tied to any one styling as long as it's wide spread adopted)

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  parserPreset: 'conventional-changelog-conventionalcommits',
+  rules: {
+    'body-leading-blank': [1, 'always'],
+    'body-max-line-length': [2, 'always', 100],
+    'footer-leading-blank': [1, 'always'],
+    'footer-max-line-length': [2, 'always', 100],
+    'header-max-length': [2, 'always', 100],
```
Comment (akapur99) [inline commitlint.config.js:8]: Shouldn't max length of header be 50 characters?

Comment (kleyow) [reply commitlint.config.js:8]: Typically yes we aim for 50. Depending on who you ask 50 is a guideline. 
It's up to us to decide how we want to move forward

Comment (eoln) [reply commitlint.config.js:8]: I would keep 100, using conventional-commits the messages are much longer

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      ['sentence-case', 'start-case', 'pascal-case', 'upper-case']
+    ],
+    'subject-empty': [2, 'never'],
+    'subject-full-stop': [2, 'never', '.'],
+    'type-case': [2, 'always', 'lower-case'],
+    'type-empty': [2, 'never'],
+    'type-enum': [
+      2,
+      'always',
+      [
```
Comment (eoln) [inline commitlint.config.js:22]: I would import this list from .versionrc file

Comment (kleyow) [reply commitlint.config.js]: done

File: commitlint.config.js (hunk 1 above)
Comment (kleyow) [reply commitlint.config.js:14]: sounds good

File: .eslintignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 node_modules/**/*.js
 coverage/*
-.circleci/*
\ No newline at end of file
+.circleci/*
+commitlint.config.js
```
Comment (eoln) [inline .eslintignore:4]: I think we need this file linted

File: commitlint.config.js [hunk 4]
```diff
@@ -1,3 +1,9 @@
+const versionrc = require('./.versionrc.js')
+
+const types = versionrc.types.map(function(typeDict) {
```
Comment (eoln) [inline commitlint.config.js:3]:
```suggestion
const types = versionrc.types.map((typeDict) => typeDict.type)
```

Comment (kleyow) [reply commitlint.config.js]: done

File: .eslintignore (hunk 3 above)
Comment (kleyow) [reply .eslintignore]: yep. don't know what I was thinking. 
the right thing was to add "typescript/no-var-requires" to eslint file
done.

**Comment by kleyow** [comment]
@eoln ping :D

**Pull request merged**